treeherder/auth/backends.py

```diff
@@ -1,83 +1,145 @@
+    def _get_jwks_json(self):
+        """
+        Get the JSON Web Key Set (jwks), which is a set of keys
+        containing the public keys that should be used to verify
+        any JWT issued by the authorization server. Auth0 exposes
+        a JWKS endpoint for each tenant, which is found at
+        'https://' + AUTH0_DOMAIN + '/.well-known/jwks.json'. This endpoint
+        will contain the JWK used to sign all Auth0 issued JWTs for this tenant.
+        Reference: https://auth0.com/docs/jwks
+        """
 
-        auth_header = request.META.get("HTTP_TCAUTH", None)
-        host = request.get_host().split(":")[0]
-        port = int(request.get_port())
+        cache_key = 'well-known-jwks'
+        cached_jwks = cache.get(cache_key)
 
-        if not auth_header:
-            # Doesn't have the right params for this backend.  So just
-            # skip and let another backend have a try at it.
-            return None
+        if cached_jwks is not None:
+            return cached_jwks
+
+        cached_jwks = urlopen('https://' + AUTH0_DOMAIN + '/.well-known/jwks.json').read()
+        cache.set(cache_key, cached_jwks)
+
+        return cached_jwks
+
+    def _validate_token(self, request):
+        token = self._get_token_auth_header(request)
+        # JWT Validator
+        # Per https://auth0.com/docs/quickstart/backend/python/01-authorization#create-the-jwt-validation-decorator
+        jwks_json = self._get_jwks_json()
+        jwks = json.loads(jwks_json)
 
-        tc_auth = Auth()
-        # https://docs.taskcluster.net/reference/platform/taskcluster-auth/references/api#authenticateHawk
-        # https://github.com/taskcluster/taskcluster-client.py#authenticate-hawk-request
-        result = tc_auth.authenticateHawk({
-            "authorization": auth_header,
-            "host": host,
-            "port": port,
-            "resource": reverse("auth-login"),
-            "method": "get",
-        })
+        unverified_header = jwt.get_unverified_header(token)
 
-        if result["status"] != "auth-success":
-            logger.warning("Error logging in: {}".format(result["message"]))
-            raise TaskclusterAuthException(result["message"])
+        rsa_key = {}
+        for key in jwks["keys"]:
+            if key["kid"] == unverified_header["kid"]:
+                rsa_key = {
+                    "kty": key["kty"],
+                    "kid": key["kid"],
+                    "use": key["use"],
+                    "n": key["n"],
+                    "e": key["e"]
+                }
 
-        client_id = result["clientId"]
-        email = self._extract_email_from_clientid(client_id)
+        if not rsa_key:
+            raise AuthError({"code": "rsa_key",
+                            "description": "rsa_key is empty"}, 401)
+
+        try:
+            jwt.decode(
+                token,
+                rsa_key,
+                algorithms=['RS256'],
+                audience=AUTH0_AUDIENCE,
```

Comment:
I think this should be `AUTH0_DOMAIN + "/userinfo"` per https://auth0.com/docs/tokens/access-token -- that is the audience for access tokens returned from an authorization with `scope=openid`.  Basically all we're doing with this token is handing it to that `/userinfo` endpoint (in `_get_user_info`), so I think that's sufficient.

Or, if verifying an id_token, it can be omitted entirely.